### PR TITLE
[ROCM] rocBLAS: default algorithm fallback

### PR DESCRIPTION
**Pull request by pemeliya**
Some number types (like complex64) are generally supported by rocBLAS but the library does not provide any solutions for autotuning. In that case, we shall fallback to use the default solution (kDefaultAlgorithm).

Besides, I have also added workspace buffer provided by XLA runtime which previously was ignored by rocBLAS.

@xla-rotation: would you please have a look ?

### CODE REVIEW

**Comment by golechwierowicz** [comment]
> Besides, I have also added workspace buffer provided by XLA runtime which previously was ignored by rocBLAS.

Can you separate this change into a separate PR?

**Comment by pemeliya** [comment]
> > Besides, I have also added workspace buffer provided by XLA runtime which previously was ignored by rocBLAS.
> 
> Can you separate this change into a separate PR?

Yes, I have created another PR for workspace settings: https://github.com/openxla/xla/pull/13779

File: xla/stream_executor/rocm/rocm_blas.cc [hunk 1]
```diff
@@ -692,17 +692,15 @@ bool ROCMBlas::GetBlasGemmAlgorithms(
...
-    solutions_.resize(num_sols);
+    // If get_solutions call fails, we still can use the default (fallback) 
+    // algorithm which is available for almost all number types.
     if (auto ret = blas_func(handle, std::forward<decltype(rest)>(rest)...,
-                             solutions_.data(), &num_sols);
-        ret != rocblas_status_success) {
-      return ret;
+                         nullptr, &num_sols); ret == rocblas_status_success) {
+      solutions_.resize(num_sols);
+      if(ret = blas_func(handle, std::forward<decltype(rest)>(rest)...,
```
Comment (golechwierowicz) [inline xla/stream_executor/rocm/rocm_blas.cc:700]: Linter will complain about the lack of space between `if` and `(ret`.

Can you add a space there?

**Review by golechwierowicz — COMMENTED** [review]
Thanks for splitting the PR!

File: xla/stream_executor/rocm/rocm_blas.cc (hunk 1 above)
Comment (pemeliya) [reply xla/stream_executor/rocm/rocm_blas.cc]: yep, fixed it

**Review by golechwierowicz — APPROVED** [review]

**Pull request closed without merging**